src/pylint_odoo/checkers/function_checker.py

```diff
@@ -0,0 +1,45 @@
+from astroid import ClassDef, FunctionDef
+from pylint.checkers import utils
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8501": (
+        "No exceptions should be raised inside unlink() functions",
+        "no-raise-unlink",
+        "Use @api.ondelete to add any constraints instead",
+    ),
+}
+
+
+class FunctionChecker(OdooBaseChecker):
+    name = "odoolint"
+    msgs = ODOO_MSGS
+    checks_maxmin_odoo_version = {"no-raise-unlink": {"odoo_minversion": "15.0"}}
+
+    @staticmethod
+    def _is_unlink(node: FunctionDef) -> bool:
+        parent = getattr(node, "parent", False)
+        return (
+            isinstance(parent, ClassDef)
+            and ("_name" in parent.locals or "_inherit" in parent.locals)
+            and node.name == "unlink"
+        )
+
+    @staticmethod
+    def get_enclosing_function(node, depth=10):
+        parent = getattr(node, "parent", False)
+        for _i in range(depth):
+            if not parent or isinstance(parent, FunctionDef):
+                return parent
```

Comment:
I think we could just break and return parent on any case after this loop, IMHO it's clearer that way.